app/workers/digest_email_generation_worker.rb

```diff
@@ -1,27 +1,30 @@
 class DigestEmailGenerationWorker
   include Sidekiq::Worker
 
-  def perform(subscriber_id:, digest_run_id:)
-    @subscriber = Subscriber.find(subscriber_id)
-    @digest_run = DigestRun.find(digest_run_id)
+  def perform(digest_run_subscriber_id)
+    @digest_run_subscriber = DigestRunSubscriber.find(digest_run_subscriber_id)
+    @subscriber = digest_run_subscriber.subscriber
+    @digest_run = digest_run_subscriber.digest_run
 
-    generate_email_and_subscription_contents
+    Email.transaction do
+      generate_email_and_subscription_contents
+
+      DeliveryRequestWorker.perform_async_with_priority(email.id, priority: :low)
 
-    DeliveryRequestWorker.perform_async_with_priority(email.id, priority: :low)
+      digest_run_subscriber.mark_completed!
```

Comment:
I think this is the point where we also want to mark a digest as completed if all digest runs are complete.

Reply:
Added :+1: